nautobot/dcim/tests/test_views.py

```diff
@@ -4761,3 +4773,61 @@ def setUpTestData(cls):
         cls.bulk_edit_data = {
             "weight": 300,
         }
+
+
+class VirtualDeviceContextTestCase(ViewTestCases.PrimaryObjectViewTestCase):
+    model = VirtualDeviceContext
+    filterset = VirtualDeviceContextFilterSet
+
+    @classmethod
+    def setUpTestData(cls):
+        devices = Device.objects.all()
+        vdc_status = Status.objects.get_for_model(VirtualDeviceContext)[0]
+        tenants = Tenant.objects.all()
+
+        cls.form_data = {
+            "name": "Virtual Device Context 1",
+            "device": devices[0].pk,
+            "identifier": 100,
+            "status": vdc_status.pk,
+            "tenant": tenants[0].pk,
+            "primary_ip4": None,
+            "primary_ip6": None,
```

Comment:
Can we set some IPs and interfaces on the selected device so that we can use more complete data here?

Reply:
According to Device model, we can't have a primary_ip for an object that does not exists, 
I implemented the same logic here.

Reply:
Device <-> Interface <-> IP <-> Device.primary_ip is a circular dependency so I can understand that restriction there. I wouldn't think the same issue would apply to VDC since the relevant device, interfaces, and IP all pre-exist in this case.

Reply:
A `primary_ips` can only be assigned after a `VDC` is created and linked to an `Interface`, as the `Interface`-`VDC` relationship is required for the `primary_ips` assignment.